test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,597 @@
+        else:
+            # value can be anything other than None
+            max_size = -1
+            match = "size should be an int or a sequence of length 1"
+
+        with pytest.raises(ValueError, match=match):
+            F.resize(self._make_input(input_type), size=size, max_size=max_size, antialias=True)
+
+    @pytest.mark.parametrize(
+        "input_type_and_kernel",
+        [
+            (datapoints.Image, F.resize_image_tensor),
+            (datapoints.Video, F.resize_video),
+        ],
+    )
+    def test_antialias_warning(self, input_type_and_kernel):
+        input_type, kernel = input_type_and_kernel
+
+        with assert_warns_antialias_default_value():
+            F.resize(self._make_input(input_type), size=self.SIZES[0])
+
+    # `InterpolationMode.NEAREST_EXACT` has no proper corresponding integer equivalent. Internally, we map it to `0` to
+    # be the same as `InterpolationMode.NEAREST` for PIL. However, for the tensor backend there is a difference and thus
+    # we don't test it here.
+    @pytest.mark.parametrize("interpolation", set(INTERPOLATION_MODES) - {transforms.InterpolationMode.NEAREST_EXACT})
+    @pytest.mark.parametrize(
+        "input_type_and_kernel",
+        [
+            (datapoints.Image, F.resize_image_tensor),
+            (datapoints.Video, F.resize_video),
+        ],
+    )
+    def test_interpolation_int(self, interpolation, input_type_and_kernel):
+        input_type, kernel = input_type_and_kernel
+        input = self._make_input(input_type)
+
+        expected = F.resize(input, size=self.SIZES[0], interpolation=interpolation)
+        actual = F.resize(
+            input,
+            size=self.SIZES[0],
+            interpolation={
+                transforms.InterpolationMode.NEAREST: 0,
+                transforms.InterpolationMode.BILINEAR: 2,
+                transforms.InterpolationMode.BICUBIC: 3,
+            }[interpolation],
+        )
+
+        assert_equal(actual, expected)
+
+    @pytest.mark.parametrize("size", SIZES)
+    @pytest.mark.parametrize("interpolation", INTERPOLATION_MODES)
+    @pytest.mark.parametrize("use_max_size", [True, False])
+    @pytest.mark.parametrize("antialias", [True, False])
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    @pytest.mark.parametrize(
+        "input_type",
+        [
+            torch.Tensor,
+            PIL.Image.Image,
+            datapoints.Image,
+            datapoints.BoundingBox,
+            datapoints.Mask,
+            datapoints.Video,
+        ],
+    )
+    def test_transform(self, size, interpolation, use_max_size, antialias, device, input_type):
+        if not (max_size_kwarg := self._make_max_size_kwarg(use_max_size=use_max_size, size=size)):
+            return
+
+        if input_type is PIL.Image.Image and antialias is False:
+            # antialias is always True for PIL
+            return
+
+        input = self._make_input(input_type, device=device)
+
+        check_transform(
```

Comment:
Being mindful of testing times, the parametrization above seems exaggerated considering the actual checks being performed by `check_transform(). A much smaller subset is probably more than enough.

Reply:
Fair enough. For the dispatcher checks, we only parametrized over `size`. Will use that here as well, but feel free to suggest a more exhaustive one.